src/qcodes/parameters/multi_channel_instrument_parameter.py

```diff
@@ -45,16 +47,33 @@ def get_raw(self) -> tuple[ParamRawDataType, ...]:
         """
         return tuple(chan.parameters[self._param_name].get() for chan in self._channels)
 
-    def set_raw(self, value: ParamRawDataType) -> None:
+    def set_raw(self, value: ParamRawDataType | Sequence[ParamRawDataType]) -> None:
         """
-        Set all parameters to this value.
+        Set all parameters to this/these value(s).
 
         Args:
-            value: The value to set to. The type is given by the
+            value: The value(s) to set to. The type is given by the
                 underlying parameter.
         """
-        for chan in self._channels:
-            getattr(chan, self._param_name).set(value)
+        try:
+            for chan in self._channels:
+                getattr(chan, self._param_name).set(value)
+        except Exception as err:
+            try:
+                for chan, val in zip(self._channels, value, strict=True):
+                    getattr(chan, self._param_name).set(val)
+            except (TypeError, ValueError):
+                note = (
+                    "Value should either be valid for a single parameter of the channel list "
+                    "or a sequence of valid values of the same length as the list."
+                )
+                try:
+                    err.add_note(note)
```

Comment:
@jenshnielsen I don't know why pyright gets offended by this line. It is in a try-except block handling exactly that error pyright tries to warn about.

Reply:
The way type checkers work they assume that the code must be valid for that python version. The way to fix that is usually to explicitly check for the version of python. See example here https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks (From the mypy docs but it works the same in pyright)

This also has be benefit that ruff/pyupgrade can automatically remove the work around code block when we drop support that the specific version